src/command/FileCommands.cxx

```diff
@@ -224,12 +230,212 @@ handle_read_comments(Client &client, Request args, Response &r)
 	case LocatedUri::Type::ABSOLUTE:
 		return read_stream_comments(r, located_uri.canonical_uri);
 
-	case LocatedUri::Type::RELATIVE:
-		return read_db_comments(client, r, located_uri.canonical_uri);
 
-	case LocatedUri::Type::PATH:
-		return read_file_comments(r, located_uri.path);
-	}
+    case LocatedUri::Type::RELATIVE:
+        return read_db_comments(client, r, located_uri.canonical_uri);
+
+    case LocatedUri::Type::PATH:
+        return read_file_comments(r, located_uri.path);
+    }
+
+    gcc_unreachable();
+}
+
+InputStreamPtr
+find_stream_art(const char *directory)
+{
+    static const std::string art_names[] = {
+        "cover.png",
+        "cover.jpg",
+        "cover.tiff",
+        "cover.bmp"
+    };
+    
+    for(const std::string& name: art_names) {
+        const std::string art_file = PathTraitsUTF8::Build(directory, name.c_str());
+        
+        try {
+            Mutex mutex;
+            Cond cond;
+            
+            auto is = InputStream::OpenReady(art_file.c_str(), mutex, cond);
+            return is;
+        } catch (const std::exception &e) {
+            continue;
```

Comment:
The intent is just to carry on with the next filename option if the `InputStream` open fails, assuming because the file isn't there. Is there a better way to handle the error? In Swift you can't have functions that throw outside a `try` block, even if that block is in a calling function.